backend/src/main/java/com/woowacourse/levellog/feedback/application/FeedbackService.java

```diff
@@ -34,99 +33,104 @@ public class FeedbackService {
+        return feedbackRepository.save(feedback).getId();
     }
 
-    public FeedbacksResponse findAll(final Long levellogId) {
+    public FeedbacksDto findAll(final Long levellogId) {
         final Levellog levellog = getLevellog(levellogId);
-        final List<FeedbackResponse> responses = getFeedbackResponses(feedbackRepository.findAllByLevellog(levellog));
+        final List<FeedbackDto> responses = getFeedbackResponses(feedbackRepository.findAllByLevellog(levellog));
 
-        return new FeedbacksResponse(responses);
+        return new FeedbacksDto(responses);
     }
 
-    public FeedbacksResponse findAllByTo(final Long memberId) {
+    public FeedbacksDto findAllByTo(final Long memberId) {
         final Member member = getMember(memberId);
         final List<Feedback> feedbacks = feedbackRepository.findAllByToOrderByUpdatedAtDesc(member);
-        return new FeedbacksResponse(getFeedbackResponses(feedbacks));
+
+        return new FeedbacksDto(getFeedbackResponses(feedbacks));
     }
 
-    public void update(final Long id, final FeedbackRequest request) {
-        final Feedback feedback = feedbackRepository.findById(id)
-                .orElseThrow(FeedbackNotFoundException::new);
+    @Transactional
+    public void update(final CreateFeedbackDto request, final Long feedbackId, final Long memberId) {
+        final Feedback feedback = getFeedback(feedbackId);
+        final Member member = getMember(memberId);
+
+        validateAuthor(feedback, member, "자신이 남긴 피드백만 수정할 수 있습니다.");
 
         feedback.updateFeedback(
                 request.getFeedback().getStudy(),
                 request.getFeedback().getSpeak(),
                 request.getFeedback().getEtc());
     }
 
-    public void deleteById(final Long id, final Long memberId) {
-        final Feedback feedback = feedbackRepository.findById(id)
-                .orElseThrow(FeedbackNotFoundException::new);
+    @Transactional
+    public void deleteById(final Long feedbackId, final Long memberId) {
+        final Feedback feedback = getFeedback(feedbackId);
+        final Member member = getMember(memberId);
 
-        final Member member = memberRepository.findById(memberId)
-                .orElseThrow(MemberNotFoundException::new);
+        validateAuthor(feedback, member, "자신이 남긴 피드백만 삭제할 수 있습니다.");
+
+        feedbackRepository.deleteById(feedbackId);
+    }
+
+    private void validateExistence(final Long levellogId, final Long fromMemberId) {
+        if (feedbackRepository.existsByLevellogIdAndFromId(levellogId, fromMemberId)) {
+            throw new FeedbackAlreadyExistException(levellogId);
+        }
+    }
+
+    private void validateTeamMember(final Long levellogId, final Member member) {
+        final Team team = getLevellog(levellogId).getTeam();
+
+        if (!participantRepository.existsByMemberAndTeam(member, team)) {
+            throw new InvalidFeedbackException("같은 팀에 속한 멤버만 피드백을 작성할 수 있습니다.");
```

Comment:
여기서도 더 많은 정보를 담는건 어떨까요?
```suggestion
            throw new InvalidFeedbackException("같은 팀에 속한 멤버만 피드백을 작성할 수 있습니다. memberId :" + memberId + " teamId : " + teamId);
```